### PR TITLE
TaskList JavaScript Functionality

### LINKED ISSUE #26: TaskList-JS Functionality (Add)
Have a feature to add notes in the TaskList

### LINKED ISSUE #27: TaskList JS Functionality (Edit)
Have a JavaScript functionality to edit the script when focused

### PR DESCRIPTION
**Pull request by KaustubhPaliwal**
## Pull Request

**Description**

Adds functionality to the TaskList Page. A detailed list of all the features:

- Allows you to add a new task with the + button
- Added the Edit and Delete Buttons
- Added the Edit Task and Add Task Modal Pages
- Allows you to enter a date and display the due date for each task
- For overdue tasks, it shows 'OVERDUE' in red text, for tasks due today, it shows 'Today', for tomorrow it says 'Tomorrow', and for the other dates it says the date in the format '10th May, 2024'
- Automatically arranges the tasks in ascending order of due date (closest task at the top)
- All 'checked off' tasks go to the button and are struckthrough, They stay there for 24 hours (while still in localStorage), after which they are 'deleted' (and removed from localStorage)
- Prints to console the total number of tasks and the completed task for the Dashboard
- Also stores a variable 'date' in local storage for the Calendar Functionality

**Type of change**
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

**Checklist**
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

**Related Issues**
Closes #26, Closes #27

### CODE REVIEW

File: src/scripts/main/public/html/task-list.html [hunk 1]
```diff
@@ -20,66 +20,47 @@ <h1>TaskList</h1>
...
-            <div class="union-wrapper">
-                <button id="openModal" class = "button-style">
-                  <img class="union" src="../img/union.svg" />
+        <div class="to-do-list">
+          
+          <div class="div">
+            <div class = "tasks-container">
+              <div class="overlap">
+                <label class="group">
+                  <input type="checkbox" id="task-checkbox" class="rectangle-checkbox">
```
Comment (jochshen) [inline src/scripts/main/public/html/task-list.html]: IDs should be camelCase

File: src/scripts/main/public/html/task-list.html [hunk 2]
```diff
@@ -20,66 +20,47 @@ <h1>TaskList</h1>
...
-                document.getElementById("popupContent").innerHTML = "";
-              });
+              </div>
+            </div>  
+          </div>
 
-            </script>
-            
+          <!--pop-up page for add tasklist starts from here-->
+          <div class="modal" id="modal">
```
Comment (jochshen) [inline src/scripts/main/public/html/task-list.html]: Try to not have classes and ids the same name, it can get confusing for CSS.

Comment (SamvathnaEm) [reply src/scripts/main/public/html/task-list.html]: Thank you for the feedback, 
I just renamed the ID to ensure it differs from the name of the class and follows the camelCase style.
Please let me know if there's anything else I need to fix.

File: src/scripts/main/public/html/task-list.html (hunk 1 above)
Comment (SamvathnaEm) [reply src/scripts/main/public/html/task-list.html]: Thank you for the feedback,
I just renamed the ID to ensure it follows the camelCase style.
Please let me know if there's anything else I need to fix.

**Review by jochshen — APPROVED** [review]
Thanks for changing the class and ids to ensure code consistency in terms of naming convention. This should be an important pr since it covers the task list javascript, which is one of the main features of our journal!

**Review by SamvathnaEm — APPROVED** [review]
The pull request is approved.

All the feedback has been addressed and adjusted accordingly. I also appreciate the implementation of the new feature.

**Pull request merged**